examples/active_learning.ipynb

```diff
@@ -0,0 +1,808 @@
+    {
+     "data": {
+      "text/html": [
+       "<pre style=\"white-space:pre;overflow-x:auto;line-height:normal;font-family:Menlo,'DejaVu Sans Mono',consolas,'Courier New',monospace\">┏━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━┓\n",
+       "┃<span style=\"font-weight: bold\">        Test metric        </span>┃<span style=\"font-weight: bold\">       DataLoader 0        </span>┃\n",
+       "┡━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━┩\n",
+       "│<span style=\"color: #008080; text-decoration-color: #008080\">  batch_averaged_test/mse  </span>│<span style=\"color: #800080; text-decoration-color: #800080\">     1.079138159751892     </span>│\n",
+       "└───────────────────────────┴───────────────────────────┘\n",
+       "</pre>\n"
+      ],
+      "text/plain": [
+       "┏━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━┓\n",
+       "┃\u001b[1m \u001b[0m\u001b[1m       Test metric       \u001b[0m\u001b[1m \u001b[0m┃\u001b[1m \u001b[0m\u001b[1m      DataLoader 0       \u001b[0m\u001b[1m \u001b[0m┃\n",
+       "┡━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━┩\n",
+       "│\u001b[36m \u001b[0m\u001b[36m batch_averaged_test/mse \u001b[0m\u001b[36m \u001b[0m│\u001b[35m \u001b[0m\u001b[35m    1.079138159751892    \u001b[0m\u001b[35m \u001b[0m│\n",
+       "└───────────────────────────┴───────────────────────────┘\n"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "trainval_data = starting_data\n",
+    "results = []\n",
+    "\n",
+    "for _ in range(len(additional_data) // query_size):\n",
+    "    # sort new datapoints by priority using priority function\n",
+    "    priority_remaining_data = [\n",
+    "        (priority_function(mpnn, datapoint), datapoint) for datapoint in additional_data\n",
+    "    ]\n",
+    "    sorted_remaining_data = [\n",
+    "        datapoint\n",
+    "        for unc, datapoint in sorted(priority_remaining_data, key=lambda d: d[0], reverse=True)\n",
+    "    ]\n",
+    "\n",
+    "    new_data = sorted_remaining_data[:query_size]\n",
+    "    additional_data = additional_data[query_size:]\n",
+    "\n",
+    "    request_labels(new_data)\n",
+    "    trainval_data.extend(new_data)\n",
+    "\n",
+    "    train_loader, val_loader, scaler = get_dataloaders(trainval_data)\n",
+    "\n",
+    "    mpnn = get_mpnn(scaler)\n",
+    "    trainer.fit(mpnn, train_loader, val_loader)\n",
+    "\n",
+    "    result = trainer.test(mpnn, test_loader)\n",
+    "    results.append((len(trainval_data), result[0][\"batch_averaged_test/mse\"]))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Finally we can view the results. The model's performance will hopefully improve with each iteration of active learning. Though this notebook is just an example. We didn't train the model for many epochs, and we used a very small dataset, so we don't expect to see the model improve. "
```

Comment:
@shihchengli What do you think about this explanation?

Reply:
It looks good!